safe_transaction_service/history/services/transaction_service.py

```diff
@@ -371,3 +373,546 @@ def serialize_all_txs_v2(
+                )
+
+                UNION ALL
+
+                -- Standalone ERC721 transfers
+                SELECT erc721.timestamp as execution_date, erc721.ethereum_tx_id as transaction_hash
+                FROM history_erc721transfer erc721
+                WHERE (erc721.to = %s OR erc721._from = %s)
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_multisigtransaction mt
+                    WHERE mt.ethereum_tx_id = erc721.ethereum_tx_id
+                )
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_moduletransaction modtx
+                    JOIN history_internaltx itx ON modtx.internal_tx_id = itx.id
+                    WHERE itx.ethereum_tx_id = erc721.ethereum_tx_id
+                )
+
+                UNION ALL
+
+                -- Standalone Ether transfers
+                SELECT itx.timestamp as execution_date, itx.ethereum_tx_id as transaction_hash
+                FROM history_internaltx itx
+                WHERE (itx.to = %s OR itx._from = %s)
+                AND itx.call_type = 0 AND itx.value > 0
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_multisigtransaction mt
+                    WHERE mt.ethereum_tx_id = itx.ethereum_tx_id
+                )
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_moduletransaction modtx
+                    JOIN history_internaltx itx2 ON modtx.internal_tx_id = itx2.id
+                    WHERE itx2.ethereum_tx_id = itx.ethereum_tx_id
+                )
+            ) combined
+        )
+        SELECT COUNT(DISTINCT (execution_date, transaction_hash))
+        FROM export_data
+        WHERE {where_clause}
+        """
+
+        # Parameters for main query (safe_address repeated for each UNION)
+        safe_address_bytes = bytes.fromhex(safe_address[2:])
+        main_params = (
+            [safe_address_bytes] * 14  # 14 instances of safe address in the query
+            + params  # date filters
+            + [limit, offset]
+        )
+
+        # Parameters for count query
+        count_params = [safe_address_bytes] * 10 + params  # date filters
+
+        with connection.cursor() as cursor:
+            # Get total count
+            cursor.execute(count_query, count_params)
+            total_count = cursor.fetchone()[0]
+
+            # Get the data
+            cursor.execute(main_query, main_params)
+            columns = [col[0] for col in cursor.description]
+            results = []
+
+            for row in cursor.fetchall():
+                row_dict = dict(zip(columns, row))
+
+                # Add '0x' prefix to hex strings and convert addresses to checksum format (except for null values)
+                if row_dict["safe_address"]:
+                    address = "0x" + row_dict["safe_address"]
+                    row_dict["safe_address"] = Web3.to_checksum_address(address)
```

Comment:
`Web3.to_checksum_address` is really slow, better use `safe_eth.eth.utils.fast_to_checksum_address`

Reply:
Done :)